### ISSUE TITLE
redshift causing flickering screen? 

### ISSUE THREAD

**Issue by rw1018** [jonls/redshift#221]
I just installed redshift (ubuntu 14.04) but its causing my screen to flicker between the red color temperature and regular temperature. Anybody know what might be causing this? 

Thanks!

**Comment by micaelbergeron** [comment]
Same problem here, on 4.0.4-2-ARCH, using nvidia binary drivers (352.09).
Either randr or vidmode method makes the flickering happen.

**Comment by inf3rno** [comment]
This is an old issue I guess: https://bbs.archlinux.org/viewtopic.php?id=153873
I had the same experience, there was 2 redshifts running. I ended one of them and everything is fine. I guess this is some sort of autostart bug.

**Comment by auxsvr** [comment]
In my case (nvidia) it is not exactly flickering, but an abrupt change of colours for some seconds.

**Comment by winny-** [comment]
I too had the same issue as @inf3rno, perhaps somebody should close this issue?

**Comment by ThornedRose** [comment]
I am having this same issue but only on some full screen apps (I've posted here #334 ). I definitely only have one instance of redshift starting (from selecting autostart / if I deselect that it doesn't start) however when I look at my task manager, I have two instances of redshift and one of redshift-gtk. If I kill one instance of redshift, it remains running (ditto if I kill redshift-gtk) but the flickering in certain full screen apps remains.

**Comment by ThornedRose** [comment]
Forgot to say I didn't have this problem at all on Openbox, LXDE or LXQT but have it in Deepin (on Arch Linux)

**Comment by jmayday** [comment]
Similar problem here. Currently I'm on Fedora 24 and sometimes screen flickers. It looks like I would stop & start redshift (color changes to blue, then slowly to red again). 

Those are redshift processes (it's not a problem of two instances being opened in parallel, right?):
```
[12:12:09] kuba:~ $ ps -A | grep -i redshift
 4705 tty2     00:00:00 redshift-gtk
 4707 tty2     00:00:00 redshift
```

Those are packages I have installed:
```
redshift.x86_64                      1.11-2.fc24                @fedora         
redshift-gtk.x86_64                  1.11-2.fc24                @fedora 
```

**Comment by kfiresmith** [comment]
Big thanks @inf3rno  - I forgot I had the KDE applet running redshift when I finally got around to setting up the ~/.config/redshift.conf and --user systemD unit file, which of course left me with two instances of redshift running and the dreaded flicker.  This thread came up on google page 1 for the problem and problem solved!

**Comment by inf3rno** [comment]
@kfiresmith You're welcome! :+1:  Can you write a short tutorial for the others?

**Comment by kfiresmith** [comment]
@inf3rno  - not really sure where to document, but it's pretty well covered by the first couple of posters in this thread - the problem was not with Redshift but with me foolishly running both the SystemD service and the KDE applet - they were both in contention with eachother.

**Comment by Machine-Hum** [comment]
If these fixes are not fixing your problem it's possible you have the same problem as me - Typically on my Windows machine I use flux, not knowing about redshift I downloaded some binaries for flux and used some install script. So Flux was actually conflicting with Redshift. Removed flux binaries from /usr/local/bin folder and my problem went away!

**Comment by Dekken** [comment]
I'm seeing this with on process on Debian 9 with an AMD Vega

Any thoughts?

**Comment by chinf** [comment]
I am seeing a problem with intermittent "flicker" - and I definitely have only one instance of redshift running and no other conflicting utilities. My problem appears as an infrequent and intermittent quick flash of default (daytime) colours before the redshifted colours are immediately reapplied. With the colour temp range set to 6500:3500 this is only really visible at night. I can provoke the flicker by repeatedly triggering colour transitions in gtk-redshift.

I have seen this problem for several months now with various versions of AMDGPU-PRO (hybrid) drivers on two different machines (both Ubuntu 16.04), one with RX 460 and the other with RX Vega. I can use either the default "randr" or "vidmode" methods to set colours, but the problem still exists.

For now my workaround is to fire off "redshift -ro" every so often via a crontab to minimise LUT changes.

**Comment by bestouff** [comment]
Wow ... a nearly 6-years old bug ! Is there any hope seeing this fixed ?

**Comment by spacekookie** [comment]
I've started having this issue on the `Linux 4.15-rc7` kernel, with an AMD graphics card (RX Vega 64). The flickers aren't caused by two instances being open. And only occur once every few minutes, sometimes not at all for ~20 minutes.

Any idea what this could be?

**Comment by Gotolei** [comment]
Also having random flickers every few minutes or so, started immediately after rebooting from kernel 4.14 to kernel 4.15. 4.15 just officially came out, so if that is the source of the problem then there'll likely be many more people running into this sometime soon.

**Comment by freed00m** [comment]
Also having this problem with R9 380 AMD card on Kernel 4.15

**Comment by zxvfxwing** [comment]
Same issue here after kernel update.

`inxi -Fx` command output :

```
System:    Host: zxvfxtie Kernel: 4.15.0-1-ARCH x86_64 bits: 64 gcc: 7.3.0 Desktop: i3 4.14.1
           Distro: Arch Linux

Graphics:  Card-1: Intel HD Graphics 530 bus-ID: 00:02.0
           Card-2: Advanced Micro Devices [AMD/ATI] Ellesmere [Radeon RX 470/480/570/580]
           bus-ID: 01:00.0
           Display Server: x11 (X.Org 1.19.6 ) drivers: intel,amdgpu (unloaded: modesetting,vesa)
           Resolution: 1920x1080@144.00hz
           OpenGL: renderer: AMD Radeon RX 480 Graphics (POLARIS10 / DRM 3.23.0 / 4.15.0-1-ARCH, LLVM 5.0.1)
```

**Comment by bestouff** [comment]
🌞 Aha I have found the cause of the problem, at least on my Ubuntu 17.10 !

Since a recent version, `gnome-shell` has its own native `redshift` integration. And I still had `gnome-shell-extension-redshift` installed, which means both implementations were walking on each other's toes, resulting in some flickering now and then.
I just uninstalled the extension and no more flickering, for now.

Hope this help

**Comment by spacekookie** [comment]
@bestouff That issue also occurs when running Redshift twice. Which isn't the case for me (or it seems OP either) though

**Comment by bestouff** [comment]
The thing is, `gnome-shell` has integrated its code, i.e. you can't find a process called `redshift` but the functionality is there. So if you see a process called `redshift` it means it's competing with `gnome-shell`. Try a `killall redshift`, it should work way better afterwards.

**Comment by Gotolei** [comment]
Thing is, not everyone is using `gnome-shell`.

I wonder if this should just be split into a different issue at this point, seems the original problem (two instances) was basically human error while the new problem of kernel 4.15 causing it to glitch isn't.

**Comment by spacekookie** [comment]
@bestouff You clearly didn't understand what I said. Thank you for explaining me how to **kill a program** 🙄

**Comment by freed00m** [comment]
@Gotolei Exactly, but I think the new issue could be dealt with right here as the title of the bug fits the profile.

**Comment by bestouff** [comment]
My comment was there just to help, don't take offense. Anyway I've created a separate issue for this: tommie-lie/gnome-shell-extension-redshift#30

**Comment by supagu** [comment]
yes i have this issue on 4.15 kernel single instance of redshift under manjaro kde edition

**Comment by unixfox** [comment]
I've the same issue after upgrading to the kernel 4.15. I'm using Plasma 5 and I've only one instance running. My GPU is Amd radeon RX480 and the driver is AMDGPU.
I tried to use another blue filter software called Iris (without redshift opened) and the same issue appeared so I think that's not specific to redshift.

**Comment by MageSlayer** [comment]
Same story when upgrading from 4.13 to 4.15+amdgpu dc under RX580.

**Comment by HadrienG2** [comment]
Same issue with kernel 4.15.2, amdgpu, and an RX 560. Screen flickers to the native color temperature shortly (I'd say a frame or two) from time to time. I'm running cinnamon 3.6.7, which AFAIK has no built-in blue-light killer. No suspicious output in dmesg or journalctl that could hint at the cause of the issue.

**Comment by MageSlayer** [comment]
I've created a bug for amdgpu here - https://bugs.freedesktop.org/show_bug.cgi?id=105083
You might be interested in putting some more details.

**Comment by Vavooon** [comment]
For those who are using pre-Vega Radeon cards:
You may try to enable AMDGPU DC (it is enabled by default only for Vega cards) by adding kernel parameter `amdgpu.dc=1`  in GRUB. After I enabled it I don't see flickering anymore.

**Comment by flak3** [comment]
@Vavooon I have been using this kernel parameter since 4.15 was released but still have flickers nonetheless.

**Comment by HadrienG2** [comment]
I still get occasional flickers with DC enabled, but it does feel like I get a lot less of them. I think I have not seen one in like, 15 minutes, they were a much more frequent occurence before.

**Comment by irgendwie** [comment]
@HadrienG2 the same for me, with DC enabled they are less frequent but still occurring.
Additionally I enabled the dc log option but in the moment of the flickering (presumably reset of the gamma value?) no event is logged.

I'm using RX580.

**Comment by ashleybone** [comment]
Interestingly, in my case I only see these flashes with AMDGPU DC _enabled_.  It does seem like the frequency of flashes has dropped with each successive 4.15 release (currently at 4.15.3).  I also see nothing in the logs even with amdgpu.dc_log=1 drm.debug=6.

**Comment by Vavooon** [comment]
I still see flickering, but they are much rare.

**Comment by Mel34** [comment]
Question for those affected (@Vavooon , @irgendwie , @HadrienG2 , @MageSlayer et al): are you using KDE Plasma?
Is it possible this is related to kwin gaining wayland code for "night mode"?

If I enter Display and monitors->Gamma in KDE systemsettings5, redshift is disabled temporarily and resumes after ~8 seconds. Does this happen for you as well?

**Comment by MageSlayer** [comment]
@Mel34 
Nope. Xfce4 is installed.

**Comment by Mel34** [comment]
@MageSlayer thank you.

FWIW, I'm experiencing flicker when rapidly alt-tabbing with amdgpu.dc=1 but not with amdgpu.dc=0 on Linux 4.15.3 rx560

**Comment by rnhmjoj** [comment]
I'm not using any DE, have kernel 4.15.2 and `amdgpu.dc=1`. I see a flickering when scrolling in the browser, randomly during video playback and in video games.
It's very likely related to hardware acceleration.

**Comment by ashleybone** [comment]
@Mel34 
I'm using plasma 5.12.1, but not wayland.  I don't see the behavior you're describing with the Gamma settings.  Entering that page doesn't change the state of redshift, whether it's on or off.

**Comment by HadrienG2** [comment]
@Mel34: I am using Cinnamon 3.6.7 on top of X11, so most likely the problem does not come from the display server/desktop environment layer of the stack (could be above, could be below).

**Comment by Vavooon** [comment]
I'm using latest KDE on X11. Entering Gamma doesn't disable Redshift (maybe it is day and there is no gamma changes applied).

**Comment by irgendwie** [comment]
@Mel34 not using a DE - only dwm as WM

**Comment by irgendwie** [comment]
@Mel34 not using a DE - only dwm as WM

**Comment by irgendwie** [comment]
@Mel34 not using a DE - only dwm as WM

**Comment by GLaDTheresCake** [comment]
Weirdly enough, running latest kde on arch, fidgeting with the gamma settings disables redshift *only* for my second monitor (1080p iiyama HDMI port) and not my first monitor (Acer 1440p Displayport) on my RX 470 8GB. The flicker also seems random between the two of them, maybe it has to do with refresh rates? My first monitor is 144 hz, my second 60.

**Comment by irgendwie** [comment]
It seems there is no flickering for me and color stays correct, when using the -o (one shot) mode - can anyone reproduce?

**Comment by flak3** [comment]
@irgendwie that's because it doesn't actually dynamically adjust the color temperature, but keeps the same color temperature all the time. Of course there's no flickering because that only appears when redshift changes the color temperature.

**Comment by irgendwie** [comment]
@flak3 I'm aware of that, but I was expecting a different behavior. Why is redshift continuously adjusting the color temperature although it isn't necessary? I mean I'd expect at nighttime redshift sets the color and "sleeps" until dawn, or occasionally checks if the color changed (from an external application) and only if it changed adapts it back to the desired value?

**Comment by C-Higgins** [comment]
Can confirm this happening on 4.15.3 using Cinnamon 3.6.7, only with amdgpu.dc=1. Does not happen if run with -o flag.

**Comment by HadrienG2** [comment]
Pinging from the [freedesktop thread](https://bugs.freedesktop.org/show_bug.cgi?id=105083): some patches to the gamma handling are being proposed for 4.17, can someone try amd-staging-drm-next or drm-next-4.17-wip from https://cgit.freedesktop.org/~agd5f/linux/?h=drm-next-4.17-wip ?

**Comment by dekay** [comment]
As I reported in #588, I have recently upgraded to Arch kernel 4.15.6-1 and it seems like the flickering has gone away for me.  I have a Radeon 560 with amdgpu.dc=1.

**Comment by C-Higgins** [comment]
I have updated to 4.15.6 and 4.15.10 in sequence and while the flickering is less frequent than on 4.15.3, it remains. Will update once 4.17 hits Fedora repos.

**Comment by supagu** [comment]
This is still happening on the 4.16 kernel.
Whenever the mouse moves the screen flickers....

**Comment by Mel34** [comment]
4.17-rc1 fixed it for me with amdgpu.dc=1

**Comment by ashleybone** [comment]
4.17.2 with amdgpu.dc=1 has fixed this issue for me.

**Comment by Zamundaaa** [comment]
Upgrading to 4.17 fixed this for me, too. Thankfully no flickering anymore!

**Comment by felipecustodio** [comment]
@Gnarflord my screen was flickering a lot, now testing without Compton the flickering is gone, even without the -o flag.

**Comment by boeremak** [comment]
I had this issue on 4.17.2. Closing one plasmoid in KDE seems to have fixed it. I have two identical taskbars on my two screens with 2 Redshift plasmoids.

**Comment by dekay** [comment]
I agreed with @Mel34 that this issue should be closed.

**Comment by mwip** [comment]
I experience this as well. Running Arch `5.5.11-arch1-1`. No DE, just Xmonad and `redshift-gtk` in the autostart.

**Comment by billop** [comment]
Getting this on Sway and AMDGPU, none of the kernel parameters mentioned here or on the Arch Wiki fixed the issue.

**Comment by inson1** [comment]
same - Linux  Mint 21.3

uninstall and killing process/es didnt help, only restart after that helped

**Comment by dradetsky** [comment]
I had this issue until I switched adjustment-method from randr to vidmode. The old arch forum thread suggested redshift was finicky, so I figured I should fiddle with some of the knobs, and that was the most obvious knob. I don't see anyone else mentioning adjustment-method here. Hopefully that's because you all already tried it & it didn't work, but...

**Comment by jnihil** [comment]
adjustment-method=vidmode fixed my issue as suggested by @dradetsky (thanks mate!).
